### PR TITLE
Add special linear

### PR DESCRIPTION
**Pull request by lpereira95**
Tests follow @SaitejaUtpala design. There's a lot of repetition between the tests of `SpecialLinear` and `GeneralLinear` that probably should be removed in the future.

### CODE REVIEW

**Comment by ninamiolane** [comment]
Nice! 🙌  thanks for re-opening this PR with only the SL files. A few comments:
- Could you make sure that the tests pass before moving the PR from "draft" to "ready for review"?
- if some tests are failing because of the randomized tests, could you let @SaitejaUtpala know?
- To avoid repeated code (coming from GL), we could use the LevelSet structure, as here:
https://github.com/geomstats/geomstats/blob/02fa5e24cad78611353f3b4e87d3f7cba6b339bc/geomstats/geometry/special_orthogonal.py#L31
basically encoding the fact that SL is a subgroup within GL.

I let @nguigs advice if he has a better idea.

**Comment by lpereira95** [comment]
Thanks for the feedback @ninamiolane!

In this case the tests are failing due to incompatibilities with tensorflow (I don't have it installed locally, which makes it a little hard to debug before the pull request - I'm using a mac M1 right now, which makes almost everything very trick to install).

I'll address the LevelStructure design soon (I completely agree with avoiding any kind of code duplication).

**Comment by ninamiolane** [comment]
@lpereira95 any update on this?
